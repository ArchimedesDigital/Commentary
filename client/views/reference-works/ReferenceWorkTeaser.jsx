import React from 'react';
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

ReferenceWorkTeaser = React.createClass({

	propTypes: {
		referenceWork: React.PropTypes.object.isRequired,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getInitialState() {
		return {
			open: false,
		};
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(baseTheme) };
	},

	toggleOpen() {
		this.setState({
			open: !this.state.open,
		});
	},

	createMarkup() {
		return {
			__html: Utils.trunc(this.props.referenceWork.description.replace(/(<([^>]+)>)/ig, ''), 120),
		};
	},

	render() {
		const referenceWork = this.props.referenceWork;
		const referenceWorkUrl = `/referenceWorks/${referenceWork.slug}`;


		return (
			<a
				className="reference-work-teaser"
				data-wow-duration="0.2s"
				href={referenceWorkUrl}
			>
				<h4 className="reference-work-title">
					{referenceWork.title}
				</h4>
				<hr />
				<span
					className="reference-work-description"
					dangerouslySetInnerHTML={this.createMarkup()}
				/>
			</a>
		);
	},

});
