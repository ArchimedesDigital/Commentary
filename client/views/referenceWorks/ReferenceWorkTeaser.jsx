import React from 'react';
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

ReferenceWorkTeaser = React.createClass({

	propTypes: {
		referenceWork: React.PropTypes.object.isRequired,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getInitialState() {
		return {
			open: false,
		};
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(baseTheme) };
	},

	toggleOpen() {
		this.setState({
			open: !this.state.open,
		});
	},

	render() {
		const referenceWork = this.props.referenceWork;
		const referenceWorkUrl = `/referenceWorks/${referenceWork.slug}`;
		let referenceWorkClass = 'referenceWork-teaser clearfix wow fadeInUp ';
		if (this.state.open) {
			referenceWorkClass += 'referenceWork-open';
		}


		return (<div
			className={referenceWorkClass}
			data-wow-duration="0.2s"
			onClick={this.toggleOpen}
  >
			<h4 className="reference-work-title">
				{referenceWork.title}
			</h4>
			<i className="mdi mdi-plus" />
			<i className="mdi mdi-minus" />
			<span className="reference-work-comment-count">({referenceWork.count} Comments)</span>
			<hr />
			<span
				className="reference-work-description"
			> {
					referenceWork.description
					|| `Quid faciat laetas segetes quo sidere terram vertere Mycenas ulmisque
					adiungere vites conveniat quae curum boum qui cultus habendo sit pecori
					apibus quanta experientia parcis, hinc canere incipiam`
			} </span>
			<a
				className="reference-work-read-more"
				href={referenceWorkUrl}
			>
				Read more
			</a>
		</div>);
	},

});
