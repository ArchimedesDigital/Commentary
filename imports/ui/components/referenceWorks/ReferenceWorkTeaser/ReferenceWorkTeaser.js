import React from 'react';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

// lib
import muiTheme from '/imports/lib/muiTheme';
import Utils from '/imports/lib/utils';

const ReferenceWorkTeaser = React.createClass({

	propTypes: {
		referenceWork: React.PropTypes.object.isRequired,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getInitialState() {
		return {
			open: false,
		};
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(muiTheme) };
	},

	toggleOpen() {
		this.setState({
			open: !this.state.open,
		});
	},

	createMarkup() {
		return {
			__html: Utils.trunc(this.props.referenceWork.description.replace(/(<([^>]+)>)/ig, ''), 140),
		};
	},

	render() {
		const { referenceWork } = this.props;

		if (!referenceWork) {
			return null;
		}

		const referenceWorkUrl = `/referenceWorks/${referenceWork.slug}`;

		return (
			<div
				className="reference-work-teaser"
				data-wow-duration="0.2s"
				href={referenceWorkUrl}
			>
				<a
					className="reference-work-title-link"
					href={referenceWorkUrl}
				>
					<h4 className="reference-work-title">
						{referenceWork.title}
					</h4>
				</a>
				<hr />
				<span
					className="reference-work-description"
					dangerouslySetInnerHTML={this.createMarkup()}
				/>

				<a
					className="reference-work-read-more-link"
					href={referenceWorkUrl}
				>
					<span className="reference-work-read-more">
						READ MORE
					</span>
				</a>
			</div>
		);
	},

});

export default ReferenceWorkTeaser;
