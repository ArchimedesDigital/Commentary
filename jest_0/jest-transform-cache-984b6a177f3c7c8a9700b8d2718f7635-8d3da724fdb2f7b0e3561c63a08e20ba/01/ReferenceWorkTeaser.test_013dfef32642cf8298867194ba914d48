/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_ReferenceWorkTeaser = require('./ReferenceWorkTeaser'); /* istanbul ignore next */var _ReferenceWorkTeaser2 = _interopRequireDefault(_ReferenceWorkTeaser);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('ReferenceWorkTeaser', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_ReferenceWorkTeaser2.default, /* istanbul ignore next */{
			referenceWork: {
				title: 'Test Work',
				slug: 'test-work',
				description: 'Quid faciat laetas segetes quo sidere terram vertere',
				link: 'http://ahcip.chs.harvard.edu/' } })).



		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: