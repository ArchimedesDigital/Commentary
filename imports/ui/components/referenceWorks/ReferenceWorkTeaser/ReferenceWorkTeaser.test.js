import React from 'react';
import renderer from 'react-test-renderer';

// component:
import ReferenceWorkTeaser from './ReferenceWorkTeaser';

describe('ReferenceWorkTeaser', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<ReferenceWorkTeaser
					referenceWork={{
						title: 'Test Work',
						slug: 'test-work',
						description: 'Quid faciat laetas segetes quo sidere terram vertere',
						link: 'http://ahcip.chs.harvard.edu/',
					}}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
