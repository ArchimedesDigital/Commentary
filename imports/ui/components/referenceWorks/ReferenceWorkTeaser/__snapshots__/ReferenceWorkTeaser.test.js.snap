// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferenceWorkTeaser renders correctly 1`] = `
<div
  className="reference-work-teaser"
  data-wow-duration="0.2s"
  href="/referenceWorks/test-work"
>
  <a
    className="reference-work-title-link"
    href="/referenceWorks/test-work"
  >
    <h4
      className="reference-work-title"
    >
      Test Work
    </h4>
  </a>
  <hr />
  <span
    className="reference-work-description"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Quid faciat laetas segetes quo sidere terram vertere",
      }
    }
  />
  <a
    className="reference-work-read-more-link"
    href="/referenceWorks/test-work"
  >
    <span
      className="reference-work-read-more"
    >
      READ MORE
    </span>
  </a>
</div>
`;
